tests/stress/stress_test.go

```diff
@@ -470,3 +471,88 @@ func TestSync_SubmitExtrinsic(t *testing.T) {
 	hashes, err := compareBlocksByNumberWithRetry(t, nodes, extInBlock.String())
 	require.NoError(t, err, hashes)
 }
+
+func Test_SubmitAndWatchExtrinsic(t *testing.T) {
+	t.Log("starting gossamer...")
+
+	// index of node to submit tx to
+	idx := 0 // TODO: randomise this
+
+	// start block producing node first
+	node, err := utils.RunGossamer(t, 0, utils.TestDir(t, utils.KeyList[0]), utils.GenesisDev, utils.ConfigNoGrandpa, true, true)
+	require.NoError(t, err)
+	nodes := []*utils.Node{node}
+
+	// Start rest of nodes
+	node, err = utils.RunGossamer(t, 1, utils.TestDir(t, utils.KeyList[1]), utils.GenesisDev, utils.ConfigNotAuthority, true, false)
+	require.NoError(t, err)
+	nodes = append(nodes, node)
+	node, err = utils.RunGossamer(t, 2, utils.TestDir(t, utils.KeyList[2]), utils.GenesisDev, utils.ConfigNotAuthority, true, false)
+	require.NoError(t, err)
+	nodes = append(nodes, node)
+
+	defer func() {
+		t.Log("going to tear down gossamer...")
+		errList := utils.StopNodes(t, nodes)
+		require.Len(t, errList, 0)
+	}()
+
+	// send tx to non-authority node
+	api, err := gsrpc.NewSubstrateAPI(fmt.Sprintf("ws://localhost:%s", nodes[idx].WSPort))
+	require.NoError(t, err)
+
+	meta, err := api.RPC.State.GetMetadataLatest()
+	require.NoError(t, err)
+
+	c, err := types.NewCall(meta, "System.remark", []byte{0xab})
+	require.NoError(t, err)
+
+	// Create the extrinsic
+	ext := types.NewExtrinsic(c)
+
+	genesisHash, err := api.RPC.Chain.GetBlockHash(0)
+	require.NoError(t, err)
+
+	rv, err := api.RPC.State.GetRuntimeVersionLatest()
+	require.NoError(t, err)
+
+	key, err := types.CreateStorageKey(meta, "System", "Account", signature.TestKeyringPairAlice.PublicKey, nil)
+	require.NoError(t, err)
+
+	var accInfo types.AccountInfo
+	ok, err := api.RPC.State.GetStorageLatest(key, &accInfo)
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	o := types.SignatureOptions{
+		BlockHash:          genesisHash,
+		Era:                types.ExtrinsicEra{IsImmortalEra: true},
+		GenesisHash:        genesisHash,
+		Nonce:              types.NewUCompactFromUInt(uint64(accInfo.Nonce)),
+		SpecVersion:        rv.SpecVersion,
+		Tip:                types.NewUCompactFromUInt(0),
+		TransactionVersion: rv.TransactionVersion,
+	}
+
+	// Sign the transaction using Alice's default account
+	err = ext.Sign(signature.TestKeyringPairAlice, o)
+	require.NoError(t, err)
+
+	extEnc, err := types.EncodeToHexString(ext)
+	require.NoError(t, err)
+
+	conn, _, err := websocket.DefaultDialer.Dial("ws://localhost:8546", nil)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	message := []byte(`{"id":1, "jsonrpc":"2.0", "method": "author_submitAndWatchExtrinsic", "params":["` + extEnc + `"]}`)
+
+	err = conn.WriteMessage(websocket.TextMessage, message)
+	require.NoError(t, err)
+
+	var result []byte
+	_, result, err = conn.ReadMessage()
+	require.NoError(t, err)
+	fmt.Println("result:", result)
```

Comment:
Is this println for debugging?  It could be useful for monitoring the results for test, since the result is json perhaps `fmt.Printf("result: %s\n", result)` so that we can read result.  

Also, since this is a subscription we expect multiple results, first with subscription ID, then other results as events happen.  So, maybe add something like:
```
 time.Sleep(time.Second)
	_, result, err = conn.ReadMessage()
	require.NoError(t, err)
	fmt.Printf("res 2 %s\n", result)
```
a few times to see the subscription messages, however this is a risk of blocking the test if something goes wrong with node and test is blocked waiting for results that may never arrive.  

Not sure of the best way to handle this, the stress tests seem useful for testing nodes and debugging, but they don't seem that good when trying to automate in CI, so I'm not sure how best to deal with this test.

Reply:
>  this is a risk of blocking the test 

I think this could be tackled by adding a timer.